NewForce-at-Mountwest/bangazonapi-team-horrohawk
# Departments ticket 7

## Proposed changes
Ticket 7 completes all endpoints to GET, GET single, POST, and PUT to the Department table/list. These additions also allow for query string parameters of filtering for greater than a specified budget and also for including all employees for the given departments, all in same list. 

## Related Issues
This ticket will function independently from other tickets once you have run the SQL script to build the database and seed data using the file "bangazonapi-team-horrohawk/TablesAndSeedData" from our code repository.

## Steps To Test
clone down the project to your computer
cd into BangazonAPI
from command line, "start BangazonAPI.sln" to open in Visual Studio
create the database and seed the data by running a quick new query with the data from our file on the Git repo at bangazonapi-team-horrohawk/TablesAndSeedData
- Launch the Build from Visual Studio
- You should see a list of Departments
- Add "?_include=employees" to see a subset list of employees for departments with employees
- Add "?_filter=budget&_gt=300000" to see only departments whose budgets are greater that $300000 and feel free to change this number to see how it may change the result
- Open your Postman desktop application and insert url from browser , showing the /department route
- Use the POST method to insert 
{
        "name": "Media Management",
        "budget": 50145
}
- See this new object appear in the database
- Change to the PUT method, add /{number of new Id here}
- Change object to 
{
        "name": "Print Management",
        "budget": 50145
}
- GET this route again and see the new change

Please Respond with any issues.

## What kind of change does this PR introduce?
Please check one of the following:
